components/brave_wallet_ui/common/async/wallet_async_handler.ts

```diff
@@ -293,6 +304,18 @@ handler.on(WalletActions.sendERC20Transfer.getType(), async (store, payload: ER2
 
 handler.on(WalletActions.approveTransaction.getType(), async (store, txInfo: TransactionInfo) => {
   const apiProxy = await getAPIProxy()
+  const hardwareAccount = await findHardwareAccountInfo(txInfo.fromAddress)
+  if (hardwareAccount && hardwareAccount.hardware) {
+    const { success, message } = await apiProxy.ethTxController.approveHardwareTransaction(txInfo.id)
+    if (success) {
+      let deviceKeyring = await apiProxy.getKeyringsByType(hardwareAccount.hardware.vendor)
+      const { v, r, s } = await deviceKeyring.signTransaction(hardwareAccount.hardware.path, message.replace('0x', ''))
+      await apiProxy.ethTxController.processLedgerSignature(txInfo.id, '0x' + v, r, s)
+      await refreshWalletInfo(store)
+      return
```

Comment:
Is it our expect behavior that we will continue to call `approveTransaction` below when we fail to run `approveHardwareTransaction` for a hardware account?

Reply:
done